test/internal/xcschemes/write_schemes_tests.bzl

```diff
@@ -0,0 +1,1651 @@
+        "colorize": attr.bool(mandatory = True),
+        "consolidation_maps": attr.string_list(mandatory = True),
+        "default_xcode_configuration": attr.string(mandatory = True),
+        "extension_point_identifiers_file": attr.string(mandatory = True),
+        "hosted_targets": attr.string(mandatory = True),
+        "include_transitive_preview_targets": attr.bool(mandatory = True),
+        "install_path": attr.string(mandatory = True),
+        "targets_args": attr.string_list_dict(mandatory = True),
+        "targets_env": attr.string(mandatory = True),
+        "workspace_directory": attr.string(mandatory = True),
+        "xcode_targets": attr.string(mandatory = True),
+        "xcscheme_infos": attr.string(mandatory = True),
+
+        # Expected
+        "expected_args": attr.string_list(mandatory = True),
+        "expected_writes": attr.string_dict(mandatory = True),
+    },
+)
+
+def write_schemes_test_suite(name):
+    """Test suite for `xcschemes_execution.write_schemes`.
+
+    Args:
+        name: The base name to be used in things created by this macro. Also the
+            name of the test suite.
+    """
+    test_names = []
+
+    def _add_test(
+            *,
+            name,
+
+            # Inputs
+            autogeneration_mode,
+            colorize = False,
+            consolidation_maps,
+            default_xcode_configuration,
+            extension_point_identifiers_file,
+            hosted_targets = [],
+            include_transitive_preview_targets = False,
+            install_path,
+            targets_args = {},
+            targets_env = {},
+            workspace_directory,
+            xcode_targets,
+            xcscheme_infos = [],
+
+            # Expected
+            expected_args,
+            expected_writes):
+        test_names.append(name)
+        write_schemes_test(
+            name = name,
+
+            # Inputs
+            autogeneration_mode = autogeneration_mode,
+            colorize = colorize,
+            consolidation_maps = consolidation_maps,
+            default_xcode_configuration = default_xcode_configuration,
+            extension_point_identifiers_file = extension_point_identifiers_file,
+            hosted_targets = json.encode(hosted_targets),
+            include_transitive_preview_targets = (
+                include_transitive_preview_targets
+            ),
+            install_path = install_path,
+            targets_args = targets_args,
+            targets_env = json.encode(targets_env),
+            workspace_directory = workspace_directory,
+            xcode_targets = json.encode(xcode_targets),
+            xcscheme_infos = json.encode(xcscheme_infos),
+
+            # Expected
+            expected_args = expected_args,
+            expected_writes = {
+                file.path: content
+                for file, content in expected_writes.items()
+            },
+        )
+
+    # @unsorted-dict-items
+    matches_xcode_targets = {
```

Comment:
FWIW, the `matches` here and `no_matches` below isn't clear to me as a reader what it means.

Reply:
Renaming and adding a comment.